### ISSUE TITLE
Indent Guide hight options: selection doesn't persist between restarts

### ISSUE THREAD

**Issue by kylegrieder** [izhangzhihao/intellij-rainbow-brackets#2597]
Your issue may already be reported!
Please search on the [issues](https://github.com/izhangzhihao/intellij-rainbow-brackets/issues) and the [document](https://github.com/izhangzhihao/intellij-rainbow-brackets/blob/2020.3/README.md) before creating one.

## Please check

- [X] I already support this project.
- [ ] If you are submitting a feature request, please do consider buy this plugin [here](https://www.jetbrains.com/shop/buy?item=P:N:PRAINBOWBRACKET:M) or donating us on [Open Collective](https://opencollective.com/intellij-rainbow-brackets) Or by AliPay/WeChatPay.
- [X] This issue/feature request is not reported before.

## Your programming languages
- golang
- php
- javascript
- c++(learning)

## Expected Behavior
* when an indent guide hight selection is made, it should persist between application restarts, just as the other options do

## Current Behavior
* when "All indent guide related" is selected, after restart the selected option reverts to "Only selected indent guide"

## Possible Solution
- my hypothesis is that,  because this customization is a paid feature, there might be some sort of race condition going on where the plugin isn't able to detect that I've paid for it when it loads, maybe because the IDE hasn't refreshed my license status in time after first starting up. or something along those lines. 

## Code snippet for reproduce (for bugs)
*see "Current Behavior"*
![image](https://user-images.githubusercontent.com/25424229/220429120-279e7fe3-bf01-4e13-86f3-a0f62cb5c10a.png)

## Your Environment

* Plugin version: 2023.1.3
* IDE & Operating System version, comment your env as below(go to "About IntelliJ IDEA" -> click the "copy" icon):

```
WebStorm 2022.3.2
Build #WS-223.8617.44, built on January 24, 2023
Licensed to Kyle Grieder
Subscription is active until June 2, 2023.
Runtime version: 17.0.5+1-b653.25 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 10
Metal Rendering is ON
Registry: debugger.new.tool.window.layout=true, ide.experimental.ui=true, ide.balloon.shadow.size=0

Non-Bundled Plugins: com.paperetto.dash (3.3), manjaro.mpb (1.7), com.intellij.ideolog (203.0.30.0), com.godwin.json.parser (1.4.1), String Manipulation (9.6.1), dev.meanmail.plugin.nginx-intellij-plugin (2022.1.1), A move tab left and right using the keyboard plugin - by momomo.com (12.550), com.karateca.jstoolbox (1.10), net.seesharpsoft.intellij.plugins.file-preview (1.6.4), com.nasller.CodeGlancePro (1.6.0), krasa.IdeaIconPack (1.14), com.pinkfloyded.idea-file-path-autocomplete.idea-file-path-autocomplete (0.4.1), name.kropp.intellij.makefile (223.8214.6), org.intellij.plugins.postcss (223.7571.117), ru.adelf.idea.dotenv (2022.3), spy-js (223.7571.203), com.intellij.database (223.8617.44), com.vermouthx.idea (1.14.2), net.seesharpsoft.intellij.plugins.csv (3.0.3-223), com.mallowigi (85.1.0), zielu.gittoolbox (500.0.4+213), mobi.hsz.idea.gitignore (4.4.4), tv.twelvetone.intellij.plugins.intellivue (1.1.22), com.draculatheme.idea (1.4.0), org.igu.plugins.nativescript (0.25.0), izhangzhihao.rainbow.brackets (2023.1.3)
```

**Comment by izhangzhihao** [comment]
So, are you currently on a trial or have you already paid for this plugin? 

This option only works if you on a trial or already paid to this plugin.

And BTW, please make sure you are not using a cracked IDE, like https://github.com/izhangzhihao/intellij-rainbow-brackets/issues/2571#issuecomment-1364544195

**Comment by kylegrieder** [comment]
@izhangzhihao I've already paid for the plugin through the jetbrains marketplace. 
![AA9E1B13-A5CF-435B-8B0D-0004A91120CD](https://user-images.githubusercontent.com/25424229/220971298-35b23b91-7db5-4e9a-8811-15e7e7decd8d.png)
![image](https://user-images.githubusercontent.com/25424229/220971507-230f0ac6-740b-43da-a03e-17f67c3805d0.png)

**Comment by izhangzhihao** [comment]
Are you still encountering this issue now? Can restarting the IDE solve it?

**Comment by maoxikun** [comment]
> Are you still encountering this issue now? Can restarting the IDE solve it?

我有同样的问题，Indent guides hight options选项设置为all ident guide related，重启idea会恢复为Only selected indent guide，上个版本插件没问题，目前是试用许可证。

IntelliJ IDEA 2022.3.2 (Ultimate Edition)
Build #IU-223.8617.56, built on January 26, 2023
Licensed to xikun mao
Subscription is active until March 25, 2023.
Evaluation purpose only.
Runtime version: 17.0.5+1-b653.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6.3
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 20
Metal Rendering is ON
Non-Bundled Plugins:
    com.markskelton.one-dark-theme (5.7.4)
    com.godwin.json.parser (1.4.2)
    Pythonid (223.8617.56)
    izhangzhihao.rainbow.brackets (2023.2.5)

Kotlin: 223-1.8.0-release-345-IJ8617.56

**Comment by izhangzhihao** [comment]
Hi, are you using this plugin in situations where the *network is unavailable*? For such cases, you should use JetBrains Account managed license and download the offline key from your account.

If you are in a *poor network*, please try this build snapshot(which increased the  license check period)

**Comment by maoxikun** [comment]
> Hi, are you using this plugin in situations where the _network is unavailable_? For such cases, you should use JetBrains Account managed license and download the offline key from your account.
> 
> If you are in a _poor network_, please try this build snapshot(which increased the license check period):

刚才装了两个虚拟机macos和win测试，win没问题，macos的Indent guides hight options选项重启idea会恢复为Only selected indent guide

**Comment by izhangzhihao** [comment]
Hi, thanks for the testing! I updated the setting logic and re-uploaded the attachment, could you please help me verify it? thanks!

**Comment by maoxikun** [comment]
> Hi, thanks for the testing! I updated the setting logic and re-uploaded the attachment, could you please help me verify it? thanks!

可以，这个是修改之后的吗，目前这个版本没问题
<img width="929" alt="image" src="https://user-images.githubusercontent.com/106460972/221409442-7acdf361-d240-46af-aa63-f89508986726.png">

**Comment by BabakAmini** [comment]
I doubt that this issue has something to do with the network status. PyCharm also checks the license at the beginning, and it succeeds when the plugin fails.

**Comment by izhangzhihao** [comment]
should be fixed in 2022.2.6, which will be publicly available within two days.
